File: sol/1262/1262.cpp
struct Solution {
    int maxSumDivThree(vector<int>& nums) {
        constexpr int M = 1e4 + 1;
        int mins[2][2] = {{M, M}, {M, M}};
        int count = 0, r;
        for (int x : nums){
            count += x;
            r = x % 3;
            if (r-- != 0){
                if (x <= mins[r][0]){
                    mins[r][1] = mins[r][0];
                    mins[r][0] = x;
                } else if (x <= mins[r][1]){
                    mins[r][1] = x;
                }
            }    
        }
        r = count % 3;
        if (r-- != 0){
            int opt = count;
            if (mins[r][0] != M){
                opt = mins[r][0];
            } 
            if (mins[1 - r][1] != M){
                opt = min(opt, mins[1 - r][0] + mins[1 - r][1]);
            }
            count -= opt;
        }
        return count;
    }
};

File: sol/1262/1262.py
class Solution:
    def maxSumDivThree(self, nums: List[int]) -> int:
        M = 10 ** 4 + 1
        minEles = [[], [M, M], [M, M]]
        count = sum(nums)
        for x in nums:
            r = x % 3
            if r != 0:
                if x <= minEles[r][0]:
                    minEles[r][0], minEles[r][1] = x, minEles[r][0]
                elif x <= minEles[r][1]:
                    minEles[r][1] = x
        r = count % 3
        if r != 0:
            opt = count
            if minEles[r][0] != M:
                opt = minEles[r][0]
            if minEles[3 - r][1] != M:
                opt = min(opt, minEles[3 - r][0] + minEles[3 - r][1])
            count -= opt
        return count
